crates/frontend/src/pages/config_version_list.rs

```diff
@@ -0,0 +1,206 @@
+use leptos::*;
+
+use chrono::{NaiveDateTime, TimeZone, Utc};
+use leptos_router::A;
+use serde_json::{json, Map, Value};
+
+use crate::components::skeleton::Skeleton;
+use crate::components::stat::Stat;
+use crate::components::table::types::TablePaginationProps;
+use crate::components::table::{types::Column, Table};
+use crate::types::{ConfigVersionListResponse, ListFilters};
+
+use crate::api::fetch_snapshots;
+
+#[component]
+pub fn config_version_list() -> impl IntoView {
+    let tenant_rs = use_context::<ReadSignal<String>>().unwrap();
+
+    // Signals for filters
+    let (filters, set_filters) = create_signal(ListFilters {
+        status: None,
+        from_date: Utc.timestamp_opt(0, 0).single(),
+        to_date: Utc.timestamp_opt(4130561031, 0).single(),
+        page: Some(1),
+        count: Some(10), // Limit of 10 items per page
+    });
+
+    let table_columns = create_memo(move |_| snapshot_table_columns(tenant_rs.get()));
+
+    let snapshots_resource: Resource<
+        (String, i64, i64),
+        Result<ConfigVersionListResponse, ()>,
+    > = create_blocking_resource(
+        move || {
+            (
+                tenant_rs.get(),
+                filters.get().page.unwrap_or(1),
+                filters.get().count.unwrap_or(10),
+            )
+        },
+        |(current_tenant, page, count)| async move {
+            match fetch_snapshots(current_tenant.to_string(), page, count).await {
+                Ok(response) => Ok(response),
+                Err(_) => Err(()),
+            }
```

Comment:
We are just doing unwraps after wrapping the response in Result, can we either handle the cases or for the error case put in the default value?